README.md

```diff
@@ -166,349 +48,49 @@ Preact supports modern browsers and IE9+:
-	@bind
-	updateText(e) {
-		this.setState({ text: e.target.value });
-	}
-	render({ }, { text }) {
-		return <input value={text} onInput={this.updateText} />;
-	}
-}
-```
-
-While this achieves much better runtime performance, it's still a lot of unnecessary code to wire up state to UI.
-
-Fortunately there is a solution, in the form of a module called [linkstate](https://github.com/developit/linkstate). Calling `linkState(component, 'text')` returns a function that accepts an Event and uses its associated value to update the given property in your component's state. Calls to `linkState()` with the same arguments are cached, so there is no performance penalty.  Here is the previous example rewritten using _Linked State_:
-
-```js
-import linkState from 'linkstate';
-
-class Foo extends Component {
-	render({ }, { text }) {
-		return <input value={text} onInput={linkState(this, 'text')} />;
-	}
-}
-```
-
-Simple and effective. It handles linking state from any input type, or an optional second parameter can be used to explicitly provide a keypath to the new state value.
-
-> **Note:** In Preact 7 and prior, `linkState()` was built right into Component. In 8.0, it was moved to a separate module. You can restore the 7.x behavior by using linkstate as a polyfill - see [the linkstate docs](https://github.com/developit/linkstate#usage).
-
-
-
-## Examples
-
-Here is a somewhat verbose Preact `<Link>` component:
-
-```js
-class Link extends Component {
-	render(props, state) {
-		return <a href={props.href}>{props.children}</a>;
-	}
-}
-```
-
-Since this is ES6/ES2015, we can further simplify:
-
-```js
-class Link extends Component {
-    render({ href, children }) {
-        return <a {...{ href, children }} />;
-    }
-}
-
-// or, for wide-open props support:
-class Link extends Component {
-    render(props) {
-        return <a {...props} />;
-    }
-}
+// create our tree:
+render(<main><h1>Hello</h1></main>, document.body);
 
-// or, as a stateless functional component:
-const Link = ({ children, ...props }) => (
-    <a {...props}>{ children }</a>
-);
+// update the tree in-place:
+render(<main><h1>Hello World!</h1></main>, document.body);
+// ^ only updates the text of our <h1>
 ```
 
-
-## Extensions
-
-It is likely that some projects based on Preact would wish to extend Component with great new functionality.
-
-Perhaps automatic connection to stores for a Flux-like architecture, or mixed-in context bindings to make it feel more like `React.createClass()`.  Just use ES2015 inheritance:
+Great, now we've built a User Interface using JSX and Preact! This approach would be tough to use for a full application though, since we have to re-render our whole app any time there is a change. This is where Components come into play: we can divide up our User Interface into nested Components, and each can update itself wherever it happens to be mounted rather than updating our whole app. Let's take a look at a simple component:
```

Comment:
Hooray! `render()` has taken our structure and output a User Interface! This approach demonstrates a simple case, but would be difficult to use as an application grows in complexity. Each change would be forced to calculate the difference between the current and updated stucture for the entire application. Components can help here – by dividing the User Interface into nested Components each can calculate their difference from their mounted point. Here's an example: